ssh/ssh.go

```diff
@@ -0,0 +1,153 @@
+package ssh
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"github.com/banzaicloud/pipeline/config"
+	"github.com/banzaicloud/pipeline/model"
+	"github.com/banzaicloud/pipeline/secret"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+	"strings"
+)
+
+var logger *logrus.Logger
+var log *logrus.Entry
+
+// Simple init for logging
+func init() {
+	logger = config.Logger()
+}
+
+type Key struct {
+	User                 string `json:"user,omitempty"`
+	Identifier           string `json:"identifier,omitempty"`
+	PublicKeyData        string `json:"publicKeyData,omitempty"`
+	PublicKeyFingerprint string `json:"publicKeyFingerprint,omitempty"`
+	PrivateKeyData       string `json:"PrivateKeyData,omitempty"`
+}
+
+// Get SSH Key from Bank Vaults
+func KeyGet(organizationID string, ClusterID uint) (sshKey Key, err error) {
+	log := logger.WithFields(logrus.Fields{"tag": "KeyGet"})
+	log.Info("Get SSH key")
+	if organizationID == "" || ClusterID == 0 {
+		log.Debugf("KeyGet organizationID: %q ClusterID: %q", organizationID, ClusterID)
+		return sshKey, fmt.Errorf("parameter missing")
+	}
+	db := model.GetDB()
+	awsProperties := &model.AmazonClusterModel{ClusterModelId: ClusterID}
+	if err := db.First(&awsProperties).Error; err != nil {
```

Comment:
No need to use '&'awsProperties because you already declared it as a pointer

Reply:
Sadly the compiler accepts literally anything  for method First because go lacks generics they had to use interface{} as a parameter type.